NumericOutcome.md

```diff
@@ -0,0 +1,561 @@
+    val (prefixDigits, startDigits, endDigits) = separatePrefix(start, end, base, numDigits)
+    
+    if (start == end) { // Special Case: Range Length 1
+        Vector(prefixDigits)
+    } else if (startDigits.forall(_ == 0) && endDigits.forall(_ == base - 1)) {
+        if (prefixDigits.nonEmpty) {
+            Vector(prefixDigits) // Total Optimization
+        } else {
+            throw new IllegalArgumentException("DLCs with only one outcome are not supported.")
+        }
+    } else if (prefixDigits.length == numDigits - 1) { // Special Case: Front Grouping = Back Grouping
+        startDigits.last.to(endDigits.last).toVector.map { lastDigit =>
+            prefixDigits :+ lastDigit
+        }
+    } else {
+      val front = frontGroupings(startDigits, base)
+      val middle = middleGrouping(startDigits.head, endDigits.head)
+      val back = backGroupings(endDigits, base)
+
+      val groupings = front ++ middle ++ back
+
+      groupings.map { digits =>
+        prefixDigits ++ digits
+      }
+    }
+}
+```
+
+## General Payout Curves
+
+### Design
+
+The goal of this specification is to enable general payout curve shapes efficiently and compactly while also ensuring that
+simpler and more common payout curves (such as a straight line for a forward contract) do not become complex
+while conforming to the generalized structure. 
+
+To this end, we let the set of all supported payout curves be the set of piecewise polynomial functions (with no continuity
+requirements between pieces).
+Since lines are polynomials, simple curves remain simple when represented in the language of piecewise polynomial functions.
+Any interesting (e.g. non-random) payout curve can be closely approximated using a cleverly constructed [polynomial interpolation](https://en.wikipedia.org/wiki/Polynomial_interpolation).
+And lastly, serializing these functions can be done compactly by providing only a few points of each polynomial piece so as to 
+enable the receiving party in the communication to interpolate the polynomials from this minimal amount of information.
+
+It is important to note however, that due to [Runge's phenomenon](https://en.wikipedia.org/wiki/Runge%27s_phenomenon), it will usually be preferable for clients to construct their payout curves
+using some choice of [spline interpolation](https://en.wikipedia.org/wiki/Spline_interpolation) instead of directly using polynomial interpolation (unless linear approximation is sufficient)
+where a spline is made up of polynomial pieces so that the resulting interpolation can be written as a piecewise polynomial one.
+
+### Curve Serialization
+
+In this section we detail the TLV serialization for a general `payout_function`.
+
+#### Version 0 payout_function
+
+1. type: ??? (`payout_function_v0`)
+2. data:
+   * [`u16`:`num_pts`]
+   * [`boolean`:`is_endpoint_1`]
+   * [`bigsize`:`event_outcome_1`]
+   * [`bigsize`:`outcome_payout_1`]
```

Comment:
I've been thinking about this more and I think I want to change this to allow payouts specified to have more precision. Obviously you can't have sub-satoshi payouts and rounding would still happen eventually but pre-maturely rounding here can reduce fidelity on other outcomes being interpolated and also breaks reproducibility when trying to use certain "on-paper equivalent" derivations of the same curve